opal/corelib/string.rb

```diff
@@ -1575,7 +1575,7 @@ def upto(stop, excl = false, &block)
     return enum_for :upto, stop, excl unless block_given?
     stop = Opal.coerce_to(stop, String, :to_str)
     %x{
-      var a, b, s = self.toString();
+      var a, b, s = self.toString(), value;
```

Comment:
It's quite similar to a return, but provided by the caller, so maybe it could be implemented similarly.
(I would guess an exception + some ID to avoid returning/breaking not far enough)